### PR TITLE
Process: Implement signal sending functions.

### PR DESCRIPTION
**Pull request by spevans**
- Implements suspend(), resume(), interrupt() and terminate().

- Add --signal-test to xdgTestHelper for testing signals.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/Process.swift [hunk 1]
```diff
@@ -507,11 +507,44 @@ open class Process: NSObject {
...
+
+    open func suspend() -> Bool {
+        guard isRunning else {
+            return false
+        }
+
+        suspendCount += 1
+        if suspendCount == 1, processIdentifier > 0 {
+            kill(processIdentifier, SIGSTOP)
+        }
```
Comment (parkera) [inline Foundation/Process.swift]: On Darwin we actually check the result of `kill` (for `0`) before incrementing the count. Same for `resume`.

File: TestFoundation/xdgTestHelper/main.swift [hunk 2]
```diff
@@ -55,15 +60,65 @@ class XDGCheck {
...
+
+if let arg = ProcessInfo.processInfo.arguments.last {
+    switch(arg) {
+    case "--xdgcheck":      XDGCheck.run()
+    case "--getcwd":        print(FileManager.default.currentDirectoryPath)
+    case "--echo-PWD":      print(ProcessInfo.processInfo.environment["PWD"] ?? "")
+    case "--signal-test":   signalTest()
+    default:                break
+    }
+}
```
Comment (parkera) [inline TestFoundation/xdgTestHelper/main.swift]: I think we're going to have to rename this helper tool at some point :)

(no need to do it now)

Comment (spevans) [reply TestFoundation/xdgTestHelper/main.swift]: Does this logic look correct: 
```
    open func suspend() -> Bool {
        guard isRunning && processIdentifier > 0 else {
            return false
        }

        if suspendCount == 0 {
            if kill(processIdentifier, SIGSTOP) == 0 {
                suspendCount += 1
            }
        } else {
            suspendCount += 1
        }
        return true
    }

    open func resume() -> Bool {
        guard isRunning && processIdentifier > 0 else {
            return true
        }

        if suspendCount == 1 {
            if kill(processIdentifier, SIGCONT) == 0 {
                suspendCount -= 1
            }
        } else {
            suspendCount -= 1
        }
        return true
    }
 ```

The tests ran OK on Linux/macOS and Darwin native so I think the counting is correct.

**Comment by spevans** [comment]
@swift-ci  please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@spevans I did some mangling around arguments in xdgTestHelper; can you rebase the patch on master once my FileManager patch finishes merging?

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@parkera This should match the behaviour of Darwin more closely now.

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test and merge

**Pull request merged**